libpod/container.go

```diff
@@ -286,33 +260,25 @@ func (c *Container) Create() (err error) {
 		return errors.Wrapf(ErrCtrExists, "container %s has already been created in runtime", c.ID())
 	}
 
-	// If using containers/storage, mount the container
-	if !c.config.RootfsFromImage {
-		// TODO implement directory-based root filesystems
-		if !c.state.Mounted {
-			return ErrNotImplemented
-		}
-	} else {
-		mountPoint, err := c.runtime.storageService.StartContainer(c.ID())
-		if err != nil {
-			return errors.Wrapf(err, "error mounting storage for container %s", c.ID())
-		}
-		c.state.Mounted = true
-		c.state.Mountpoint = mountPoint
-
-		logrus.Debugf("Created root filesystem for container %s at %s", c.ID(), c.state.Mountpoint)
+	mountPoint, err := c.runtime.storageService.StartContainer(c.ID())
+	if err != nil {
+		return errors.Wrapf(err, "error mounting storage for container %s", c.ID())
+	}
+	c.state.Mounted = true
+	c.state.Mountpoint = mountPoint
 
-		defer func() {
-			if err != nil {
-				if err2 := c.runtime.storageService.StopContainer(c.ID()); err2 != nil {
-					logrus.Errorf("Error unmounting storage for container %s: %v", c.ID(), err2)
-				}
+	logrus.Debugf("Created root filesystem for container %s at %s", c.ID(), c.state.Mountpoint)
 
-				c.state.Mounted = false
-				c.state.Mountpoint = ""
+	defer func() {
```

Comment:
I belive you need to pass in err

defer func() err error

Reply:
Not necessary - the error returned by Create() is named err, so this will always be the error returned by the function

Reply:
I agree.